valheim-server

```diff
@@ -23,8 +22,27 @@ main() {
         fi
     done
     echo "Running Valheim Server"
-    "$valheim_server" -nographics -batchmode -name "$SERVER_NAME" -port $SERVER_PORT -world "$WORLD_NAME" -password "$SERVER_PASS" -public $SERVER_PUBLIC &
-    valheim_server_pid=$!
+    if [ $VALHEIM_PLUS_ENABLED = true ]; then
+        export DOORSTOP_ENABLE=TRUE
+        export DOORSTOP_INVOKE_DLL_PATH="$valheim_server_dir/BepInEx/core/BepInEx.Preloader.dll"
+        doorstop_libs="$valheim_server_dir/doorstop_libs"
+        arch="x64"
+        lib_postfix="so"
+        export LD_LIBRARY_PATH=./linux64:$LD_LIBRARY_PATH
+        doorstop_libname=libdoorstop_${arch}.${lib_postfix}
+        export LD_LIBRARY_PATH="${doorstop_libs}":${LD_LIBRARY_PATH}
+        export LD_PRELOAD=$doorstop_libname:$LD_PRELOAD
+        export DYLD_LIBRARY_PATH="${doorstop_libs}"
+        export DYLD_INSERT_LIBRARIES="${doorstop_libs}/$doorstop_libname"
+        export templdpath=$LD_LIBRARY_PATH
+        export SteamAppId=892970
+    else
```

Comment:
This entire else case is unnecessary. The values are the same for Valheim vanilla and Valheim Plus. V+ just extends the LD_LIBRARY_PATH. Also there is no need to restore the old LD_LIBRARY_PATH as this script is never run from a shell where restoring the original state would be of interest.

Reply:
Yeah, I just wanted to insert the original code found in the `run_bepinex.sh` script and go from there. However, I will refactor this section to be less redundant